src/Integration/RequestIntegration.php

```diff
@@ -196,4 +184,25 @@ private function captureRequestBody(ServerRequestInterface $serverRequest)
 
         return $requestBody->getContents();
     }
+
+    private function parseUploadedFiles(array $uploadedFiles): array
+    {
+        $data = [];
+
+        foreach ($uploadedFiles as $key => $item) {
+            if ($item instanceof UploadedFileInterface) {
+                $data[$key] = [
+                    'client_filename' => $item->getClientFilename(),
+                    'client_media_type' => $item->getClientMediaType(),
+                    'size' => $item->getSize(),
+                ];
+            } elseif (\is_array($item)) {
+                $data[$key] = $this->parseUploadedFiles($item);
+            } else {
+                throw new \InvalidArgumentException('Unrecognized UploadedFiles item, got ' . \gettype($item));
```

Comment:
You should use both `gettype` and `get_class` to be sure that we handle both scalar values and objects. I would also change the exception class to `UnexpectedValueException` and the message to `Expected an object implementing the Psr\Http\Message\UploadedFileInterface interface. Got: <value type>.`